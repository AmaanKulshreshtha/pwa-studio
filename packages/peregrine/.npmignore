.eslintrc.js
.storybook
/{dist,src}/**/__{docs,helpers,mocks,stories,tests}__/**
/scripts
