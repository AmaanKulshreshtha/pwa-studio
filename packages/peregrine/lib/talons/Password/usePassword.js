import { useState, useCallback } from 'react';

export const usePassword = () => {
    const [visible, setVisbility] = useState(false);

    const togglePasswordVisibility = useCallback(() => {
        setVisbility(!visible);
    }, [visible]);

    return {
        visible,
        togglePasswordVisibility
    };
};
