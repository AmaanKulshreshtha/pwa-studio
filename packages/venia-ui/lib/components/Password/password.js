import React from 'react';
import { string, bool, shape } from 'prop-types';
import { Eye, EyeOff } from 'react-feather';

import { mergeClasses } from '@magento/venia-ui/lib/classify';
import { usePassword } from '@magento/peregrine/lib/talons/Password/usePassword';

import Button from '../Button';
import Field from '../Field';
import TextInput from '../TextInput';
import { isRequired } from '../../util/formValidators';

import defaultClasses from './password.css';

const Password = props => {
    const {
        classes: propClasses,
        label,
        fieldName,
        isToggleButtonHidden
    } = props;
    const talonProps = usePassword();
    const { visible, togglePasswordVisibility } = talonProps;
    const classes = mergeClasses(defaultClasses, propClasses);

    const passwordButton = (
        <Button
            className={classes.passwordButton}
            onClick={togglePasswordVisibility}
            type="button"
        >
            {visible ? <EyeOff /> : <Eye />}
        </Button>
    );

    const fieldType = visible ? 'text' : 'password';

    return (
        <Field label={label} classes={{ root: classes.root }}>
            <TextInput
                field={fieldName}
                type={fieldType}
                validate={isRequired}
                after={!isToggleButtonHidden && passwordButton}
            />
        </Field>
    );
};

Password.propTypes = {
    classes: shape({
        root: string
    }),
    label: string,
    fieldName: string,
    isToggleButtonHidden: bool
};

Password.defaultProps = {
    isToggleButtonHidden: true
};

export default Password;
