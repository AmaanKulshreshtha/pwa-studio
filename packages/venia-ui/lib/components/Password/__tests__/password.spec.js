import React from 'react';
import { createTestInstance } from '@magento/peregrine';

import Password from '../password';

test('should render properly', () => {
    const tree = createTestInstance(
        <Password
            label="Password"
            fieldName="password"
            isToggleButtonHidden={true}
        />
    );
    expect(tree.toJSON()).toMatchSnapshot();
});

test('should render toggle button if isToggleButtonHidden is false', () => {
    const tree = createTestInstance(
        <Password
            label="Password"
            fieldName="password"
            isToggleButtonHidden={false}
        />
    );
    expect(tree.toJSON()).toMatchSnapshot();
});
