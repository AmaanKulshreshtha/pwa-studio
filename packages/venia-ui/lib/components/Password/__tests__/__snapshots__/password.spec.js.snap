// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<div>
  <label>
    Password
  </label>
  <span
    style={
      Object {
        "--iconsAfter": 0,
        "--iconsBefore": 0,
      }
    }
  >
    <span>
      <input
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        type="password"
        value=""
      />
    </span>
    <span />
    <span />
  </span>
  <p />
</div>
`;

exports[`should render toggle button if isToggleButtonHidden is false 1`] = `
<div>
  <label>
    Password
  </label>
  <span
    style={
      Object {
        "--iconsAfter": 1,
        "--iconsBefore": 0,
      }
    }
  >
    <span>
      <input
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        type="password"
        value=""
      />
    </span>
    <span />
    <span>
      <button
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span>
          <svg
            fill="none"
            height="24"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z"
            />
            <circle
              cx="12"
              cy="12"
              r="3"
            />
          </svg>
        </span>
      </button>
    </span>
  </span>
  <p />
</div>
`;
